agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java

```diff
@@ -17,22 +18,40 @@
  */
 public final class MulticastUtil {
 
-    // IPv4 Address for Jolokia's Multicast group
-    public static final String JOLOKIA_MULTICAST_GROUP = "239.192.48.84";
+    // Default IPv4 Address for Jolokia's Multicast group
+    public static final String DEFAULT_JOLOKIA_MULTICAST_GROUP = "239.192.48.84";
 
-    // Multicast port where to listen for queries
-    public static final int JOLOKIA_MULTICAST_PORT = 24884;
+    // Default multicast port where to listen for queries
+    public static final int DEFAULT_JOLOKIA_MULTICAST_PORT = 24884;
 
     // Utility class
     private MulticastUtil() {
     }
 
+    private static <T> T getOrDefault(ConfigKey configKey, String sysEnvKey, T defaultValue) {
+        String property = System.getProperty("jolokia." + configKey.getKeyValue());
+        if (property == null) {
+            property = System.getenv(sysEnvKey);
+        }
+        if (property != null) {
+            if (defaultValue instanceof Integer) {
+               return (T) Integer.valueOf(property);
+            } else {
+                return (T) property;
+            }
+        }
+        return defaultValue;
+    }
+
     static MulticastSocket newMulticastSocket(InetAddress pAddress, LogHandler pLogHandler) throws IOException {
+        String multicastGroup = getOrDefault(ConfigKey.MULTICAST_GROUP, "JOLOKIA_MULTICAST_GROUP", DEFAULT_JOLOKIA_MULTICAST_GROUP);
```

Comment:
This utility class is not the proper place for looking up configuration values. Instead this method should have two addition parameters, the group and the port. And feed in by the caller of this method.

Reply:
So the [method initializing the multicast thread](https://github.com/rhuss/jolokia/blob/ec1cebda618e68d12c9b1906820564408cf950e1/agent/core/src/main/java/org/jolokia/http/AgentServlet.java#L198) should pass through the `Configuration` object to the `MulticastSocketListenerThread` where it then be used to call this utility method.

I know its a bit tedious, but it helps in keeping the configuration handling separate.